app/src/main/java/com/es/findsoccerplayers/AccountActivity.java

```diff
@@ -0,0 +1,96 @@
+package com.es.findsoccerplayers;
+
+import androidx.annotation.NonNull;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.appcompat.widget.Toolbar;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.Button;
+import android.widget.ImageView;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import com.google.android.gms.auth.api.signin.GoogleSignIn;
+import com.google.android.gms.auth.api.signin.GoogleSignInClient;
+import com.google.android.gms.auth.api.signin.GoogleSignInOptions;
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseUser;
+
+public class AccountActivity extends AppCompatActivity {
+
+    private Button btnLogOut;
+    private FirebaseAuth mAuth;
+    private FirebaseUser acct;
+    private GoogleSignInClient mGoogleSignInClient;
+    private TextView id;
+    private TextView email;
+    private TextView nome;
+    private ImageView photo;
+    private Toolbar toolbar;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_account);
+
+        btnLogOut=findViewById(R.id.btnLogOut);
+        mAuth=FirebaseAuth.getInstance();
+        acct = FirebaseAuth.getInstance().getCurrentUser();
+        id = (TextView) findViewById(R.id.id);
+        email = (TextView) findViewById(R.id.email);
+        nome = (TextView) findViewById(R.id.nome);
+        toolbar = findViewById(R.id.toolbar);
+        setSupportActionBar(toolbar);
+
+        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+
+        btnLogOut.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                signOut();
+                Intent intent=new Intent(AccountActivity.this, MainActivity.class);
+                startActivity(intent);
+            }
+        });
+
+        GoogleSignInOptions gso = new GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
+                .requestIdToken(getString(R.string.default_web_client_id))
+                .requestEmail()
+                .build();
+
+        mGoogleSignInClient = GoogleSignIn.getClient(this, gso);
+    }
+
+    @Override
+    protected void onStart() {
+        super.onStart();
+        id.setText("ID: "+acct.getUid());
+        email.setText("email: "+acct.getEmail());
+        nome.setText("nome: "+acct.getDisplayName());
+    }
+
+    private void signOut() {
+        mAuth.signOut();
+        mGoogleSignInClient.signOut()
```

Comment:
ma se uno non si è autenticato con google, il sign out di google funziona lo stesso?

Reply:
bah non mi ha mai dato problemi, non saprei dirti, il log out sia con google che con un account "semplice" ha sempre funzionato

Reply:
Ok della docs dicono "Signs out the current signed-in user if any.", perciò se non c'è loggato nessuno funziona lo stesso.